### PR TITLE
wasm: build action tests

### PR DESCRIPTION
**Pull request by TalDerei**
references https://github.com/penumbra-zone/web/issues/1830

### CODE REVIEW

File: packages/wasm/crate/tests/test_build.rs [hunk 1]
```diff
@@ -1,577 +1,398 @@
...
+    let spendable_note_json = r#"
+    {
+        "note_commitment": {
+            "inner": "MY7PmcrH4fhjFOoMIKEdF+x9EUhZ9CS/CIfVco7Y5wU="
+        },
+        "note": {
+            "value": {
+                "amount": {
+                    "lo": "1000000",
+                    "hi": "0"
```
Comment (grod220) [inline packages/wasm/crate/tests/test_build.rs:109]: Maintaining these strings manually feels like a chore. Can we just test the inner function of the build and not feel like we need to test the deserialization logic?

File: packages/wasm/crate/tests/utils/sct.rs
Comment (grod220) [inline packages/wasm/crate/tests/utils/sct.rs:1]: Can't we import these from core?

Comment (TalDerei) [reply packages/wasm/crate/tests/utils/sct.rs:1]: Some of the internal fields are private, which is why I emulated the types to match their structure.

File: packages/wasm/crate/tests/test_build.rs (hunk 1 above)
Comment (TalDerei) [reply packages/wasm/crate/tests/test_build.rs:109]: replaced the hardcoded json with their corresponding typed structures.

File: packages/wasm/crate/tests/test_build.rs [hunk 2]
```diff
@@ -1,577 +1,374 @@
...
+
+    // Execute parallel spend transaction and generate proof.
+    let parallel_transaction = build_parallel_inner(
+        actions,
+        transaction_plan.clone(),
+        witness_data.clone(),
+        authorization_data.clone(),
+    )
+    .unwrap();
+    console_log!("Parallel transaction is: {:?}", parallel_transaction);
```
Comment (grod220) [inline packages/wasm/crate/tests/test_build.rs:355]: nit: we probably don't want these console logs right?

File: packages/wasm/crate/tests/utils/planner_setup.rs [hunk 3]
```diff
@@ -9,6 +9,7 @@ use penumbra_wasm::database::interface::Database;
 use penumbra_wasm::database::mock::MockDb;
 use penumbra_wasm::storage::{byte_array_to_base64, Tables};
 
+#[allow(dead_code)]
```
Comment (grod220) [inline packages/wasm/crate/tests/utils/planner_setup.rs:12]: question: should we just delete this?

File: packages/wasm/crate/src/database/indexed_db.rs [hunk 4]
```diff
@@ -16,59 +15,10 @@ pub async fn open_idb_database(constants: &DbConstants) -> WasmResult<IdbDatabas
     #[allow(unused_mut)]
     let mut db_req: OpenDbRequest = IdbDatabase::open_u32(&constants.name, constants.version)?;
 
-    // Conditionally mock sample `IdbDatabase` database for testing purposes
-    #[cfg(feature = "mock-database")]
-    let db_req = mock_test_database(db_req).into_future().await;
-
```
Comment (grod220) [inline packages/wasm/crate/src/database/indexed_db.rs:22]: It feels very nice deleting this

**Review by grod220 — APPROVED** [review]
Great job!

File: packages/wasm/crate/tests/test_build.rs (hunk 2 above)
Comment (TalDerei) [reply packages/wasm/crate/tests/test_build.rs:355]: removed

File: packages/wasm/crate/tests/utils/planner_setup.rs (hunk 3 above)
Comment (TalDerei) [reply packages/wasm/crate/tests/utils/planner_setup.rs:12]: removed

**Pull request merged**